File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        maxi=0
        while left < right :
            area = (right-left)*min(height[left], height[right])
            if area > maxi:
                maxi = area
            if height[left] > height[right]:
                right -= 1
            elif height[left] <= height[right]:
                left += 1
        return maxi

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0 , end = height.size()-1 , max=0; 
        while(i<end){
            if(min(height[i],height[end])*(end-i)>max) max= min(height[i],height[end])*(end-i) ;
            if(height[i]<height[end]) i++;
            else if(height[i]>=height[end]) end--;
        }
        return max;
    }
};